### PR TITLE
Refactor and fixes

### PR DESCRIPTION
**Pull request by draagc-iw**

### CODE REVIEW

File: src/components/pages/MainPage.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    };
+
+    Hub.listen('auth', authHandler);
+
+    const createEnvironment = async () => {
+        setIsLoading(true);
+        const user = await Auth.currentAuthenticatedUser({bypassCache: true});
+        const jwtToken = user.signInUserSession.idToken.jwtToken;
+        const res = await fetch(`/connect?id_token=${jwtToken}`);
+        if (res.status === 200) return nav.go(Pages.CONNECT, {desktopUrl: res.body})
```
Comment (IanUtley) [inline src/components/pages/MainPage.js:40]: Need to add ?token= (jwtToken) to desktop URL (a known bug).

Comment (draagc) [reply src/components/pages/MainPage.js:40]: On it

File: src/components/pages/SetupMFAPage.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import {getConfigOrDefault} from "../../utils/appConfig";
+
+const SetupMFAPage = ({nav}) => {
+    const [enteredCode, setEnteredCode] = useState('');
+    const [qrCode, setQrCode] = useState();
+
+    useEffect(() => {
+        async function setupMfa() {
+            const user = await Auth.currentAuthenticatedUser();
+            const tOtpCode = await Auth.setupTOTP(user)
```
Comment (IanUtley) [inline src/components/pages/SetupMFAPage.js:15]: wierd casing

File: src/components/pages/MainPage.js (hunk 1 above)
Comment (draagc-iw) [reply src/components/pages/MainPage.js:40]: On it

**Pull request merged**